src/Libraries/Microsoft.Extensions.Http.AutoClient/AutoClientUtilities.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Extensions.Http.AutoClient;
+
+/// <summary>
+/// Utilities for AutoClient feature.
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class AutoClientUtilities
+{
+    private static readonly char[] _slashes = { '/', '\\' };
+    private static readonly char[] _slashesOrDot = { '/', '\\', '.' };
+
+    /// <summary>
+    /// Returns whether a value is a valid path argument.
+    /// </summary>
+    /// <param name="value">The value to validate.</param>
+    /// <returns>Whether the value is a valid path argument.</returns>
+    public static bool IsPathParameterValid(string value)
+    {
+        var trimmed = value.AsSpan().Trim();
+
+        if (trimmed.Length == 0)
+        {
+            return false;
+        }
+
+        // Fast path for most common cases
+        var index = trimmed.IndexOfAny(_slashesOrDot);
+        if (index == -1)
+        {
+            return true;
+        }
+
+        // Slashes can't be used
+        if (trimmed.Slice(index).IndexOfAny(_slashes) >= 0)
```

Comment:
I think you want:

trimmed = trimmed.Slice();

as a separate statement, such that it applies to both this use of IndexOfAny as well as for the following code below

Reply:
Did that on purpose or a string like `"a.."` would return `false` while it's fine.

Reply:
Ah, I see.